core/services/bulletprooftxmanager/transmitchecker.go

```diff
@@ -0,0 +1,156 @@
+package bulletprooftxmanager
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/vrf_coordinator_v2"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// simulationTimeout must be short since simulation adds latency to broadcasting a tx which can
+// negatively affect response time.
+const simulationTimeout = 2 * time.Second
+
+var (
+	_ TransmitChecker = NoChecker{}
+	_ TransmitChecker = SimulateChecker{}
+)
+
+// TransmitChecker determines whether a transaction should be submitted on-chain.
+type TransmitChecker interface {
+	ShouldTransmit(
+		ctx context.Context,
+		l logger.Logger,
+		tx EthTx,
+		a EthTxAttempt,
+	) (bool, error)
+}
+
+func buildChecker(meta *EthTxMeta, client eth.Client) (TransmitChecker, error) {
+	if meta == nil {
+		return NoChecker{}, nil
+	}
+
+	switch meta.CheckerType {
+	case TransmitCheckerTypeSimulate:
+		return SimulateChecker{client}, nil
+	case TransmitCheckerTypeVRFV2:
+		coord, err := vrf_coordinator_v2.NewVRFCoordinatorV2(meta.VRFCoordinatorAddress, client)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create coordinator at address %v", meta.VRFCoordinatorAddress)
+		}
+		return VRFV2Checker{coord.GetCommitment}, nil
+	case "":
+		return NoChecker{}, nil
+	default:
+		return nil, errors.Errorf("unrecognized checker type: %s", meta.CheckerType)
+	}
+}
+
+// NoChecker is a TransmitChecker that always determines a transaction should be submitted.
+type NoChecker struct{}
+
+func (_ NoChecker) ShouldTransmit(
+	_ context.Context,
+	_ logger.Logger,
+	_ EthTx,
+	_ EthTxAttempt,
+) (bool, error) {
+	return true, nil
+}
+
+type SimulateChecker struct {
+	Client eth.Client
+}
+
+func (s SimulateChecker) ShouldTransmit(
+	ctx context.Context,
+	l logger.Logger,
+	tx EthTx,
+	a EthTxAttempt,
+) (bool, error) {
+	simulationCtx, cancel := context.WithTimeout(ctx, simulationTimeout)
+	defer cancel()
```

Comment:
I think its reasonable to dump this timeout and rely on the context entirely

Reply:
Yeah that makes sense, I ended up moving the timeout into eth_broadcaster so it doesn't need to be reimplemented for both simulate and VRF, lemme know if this looks ok